api_fhir_r4/converters/groupConverter.py

```diff
@@ -141,45 +149,177 @@ def build_fhir_member(cls,fhir_family,imis_family):
+    def build_fhir_extentions(cls, fhir_family, imis_family, reference_type):
+        fhir_family.extension = []
+
+        def build_extension(fhir_family, imis_family, value):
+            extension = Extension.construct()
+            if value == "group-address":
+                extension.url = f"{GeneralConfiguration.get_system_base_url()}StructureDefinition/group-address"
+                family_address = cls.build_fhir_address(imis_family.address, "home", "physical")
+                if imis_family.location:
+                    family_address.state = imis_family.location.parent.parent.parent.name
+                    family_address.district = imis_family.location.parent.parent.name
+                    # municipality extension
+                    extension_address = Extension.construct()
+                    extension_address.url = f"{GeneralConfiguration.get_system_base_url()}StructureDefinition/address-municipality"
+                    extension_address.valueString = imis_family.location.parent.name
+                    family_address.extension = [extension_address]
+
+                    # address location reference extension
+                    extension_address = Extension.construct()
+                    extension_address.url = f"{GeneralConfiguration.get_system_base_url()}StructureDefinition/address-location-reference"
+                    reference_location = Reference.construct()
+                    reference_location.reference = F"Location/{imis_family.location.name}-village"
+                    extension_address.valueReference = reference_location
+                    family_address.extension.append(extension_address)
+
+                    family_address.city = imis_family.location.name
+
+                extension.valueAddress = family_address
+
+            elif value == "group-poverty-status":
+                extension.url = f"{GeneralConfiguration.get_system_base_url()}StructureDefinition/group-poverty-status"
+                extension.valueBoolean = imis_family.poverty
+
+            elif value == "group-type":
+                extension.url = f"{GeneralConfiguration.get_system_base_url()}StructureDefinition/group-type"
+                if hasattr(imis_family, "family_type") and imis_family.family_type is not None:
+                    display = GroupTypeMapping.group_type[str(imis_family.family_type.code)]
+                    system = f"{GeneralConfiguration.get_system_base_url()}CodeSystem/group-types"
+                    extension.valueCodeableConcept = cls.build_codeable_concept(code=str(imis_family.family_type.code),
+                                                                                system=system)
+                    if len(extension.valueCodeableConcept.coding) == 1:
+                        extension.valueCodeableConcept.coding[0].display = display
+            # group-confirmation
+            else:
+                nested_extension = Extension.construct()
```

Comment:
might be more readable if this will be in  a dedicated fuction

Reply:
ok

Reply:
I will fix this to have this in separate function